beetsplug/spotify.py

```diff
@@ -536,3 +556,71 @@ def _output_match_results(self, results):
             self._log.warning(
                 f'No {self.data_source} tracks found from beets query'
             )
+
+    def _fetch_info(self, items, write, force):
+        """Obtain track information from Spotify."""
+        spotify_audio_features = {
+            'acousticness': ['spotify_track_acousticness'],
+            'danceability': ['spotify_track_danceability'],
+            'energy': ['spotify_track_energy'],
+            'instrumentalness': ['spotify_track_instrumentalness'],
+            'key': ['spotify_track_key'],
+            'liveness': ['spotify_track_liveness'],
+            'loudness': ['spotify_track_loudness'],
+            'mode': ['spotify_track_mode'],
+            'speechiness': ['spotify_track_speechiness'],
+            'tempo': ['spotify_track_tempo'],
+            'time_signature': ['spotify_track_time_sig'],
+            'valence': ['spotify_track_valence'],
+        }
+        import time
+
+        no_items = len(items)
+        self._log.info('Total {} tracks', no_items)
+
+        for index, item in enumerate(items, start=1):
+            time.sleep(.5)
+            self._log.info('Processing {}/{} tracks - {} ',
+                           index, no_items, item)
```

Comment:
I think it's OK to use `len(items)` here; it's unlikely to cause measurable performance problems.